src/table/hooks/use-view-service.ts

```diff
@@ -1,16 +1,23 @@
-import { useMemo } from '@nebula.js/stardust';
-import type { ViewService } from '../../types';
+import { useMemo, useOptions } from '@nebula.js/stardust';
+import type { SnapshotData, ViewService, ViewState, UseOptions } from '../../types';
 
-const createViewService = (): ViewService => {
+const createViewService = (viewState: ViewState, snapshotData?: SnapshotData): ViewService => {
```

Comment:
`snapshotData: SnapshotData | undefined`  is more precise in this case, as snapshotData is not really optional but could be undefined

when `snapshotData?: SnapshotData`, `createViewService(viewState)` will not complained by ts